subprojects/packagefiles/theora/meson.build

```diff
@@ -0,0 +1,269 @@
+project(
+  'theora',
+  'c',
+  'cpp',
+  version: '1.1.1',
+  meson_version: '>= 0.59.0',
+  license: 'BSD-3-Clause',
+  default_options: ['c_std=c11'],
+)
+
+conf = configuration_data()
+
+enable_valgrind = get_option('valgrind-testing').allowed()
+
+test_env = []
+
+if enable_valgrind
+  valgrind_exe = find_program('valgrind')
+
+  conf.set('HAVE_VALGRIND', valgrind_exe.found())
+
+  if conf.get('HAVE_VALGRIND')
+    test_env += [
+      '-q',
+      '--leak-check=yes',
+      '--show-reachable=yes',
+      '--num-callers=100',
+    ]
+
+    message(
+      'Tests will be run under: @0@'.format(env.get('VALGRIND_ENVIRONMENT')),
+    )
+  else
+    message('Valgrind not found')
+  endif
+else
+  message('Valgrind testing not enabled')
+endif
+
+compile_args = []
+
+cc = meson.get_compiler('c')
+
+if cc.get_argument_syntax() != 'msvc'
+  compile_args += cc.get_supported_arguments('-Wall', '-Wno-parentheses')
+endif
+
+m_dep = cc.find_library('m', required: false)
+
+if get_option('collect-metrics').enabled()
+  add_project_arguments('-DOC_COLLECT_METRICS', language: 'c')
+endif
```

Comment:
This should be a boolean, it's either true or false and it's confusing if `-Dcollect-metrics=auto` and `-Dcollect-metrics=enabled` do exactly the same and both mean "unconditionally enable metrics".

Reply:
I'm using feature flags because I get Meson's boolean interpretation always mixed up.

e.g. your change would imply we'd have `-Dcollect-metrics=true` and also `-Dtelemetry=enabled`, but Meson won't recognise `enabled` for its obvious meaning and error out if it's assigned to a boolean.

Reply:
> it's confusing if -Dcollect-metrics=auto and -Dcollect-metrics=enabled do exactly the same

As per https://mesonbuild.com/Reference-manual_returned_feature.html#featureenabled `feature.enabled()` returns false if it's set to `auto`. Did you mean `feature.allowed()`?

Reply:
Err, probably.

You could equally say that if you use `if get_option('collect-metrics').enabled()` then passing:
- `-Dcollect-metrics=auto`, and 
- `-Dcollect-metrics=disabled`

do exactly the same thing. The general point remains the same, that this isn't much of a tristate option if you merge the meaning of two states.